src/slic3r/GUI/Preferences.cpp

```diff
@@ -377,29 +408,72 @@ wxBoxSizer *PreferencesDialog::create_item_region_combobox(wxString title, wxWin
     return m_sizer_combox;
 }
 
-wxBoxSizer *PreferencesDialog::create_item_loglevel_combobox(wxString title, wxWindow *parent, wxString tooltip, std::vector<wxString> vlist)
+wxBoxSizer *PreferencesDialog::create_item_autoflush(wxString title, wxString tooltip)
 {
     wxBoxSizer *m_sizer_combox = new wxBoxSizer(wxHORIZONTAL);
-    m_sizer_combox->Add(0, 0, 0, wxEXPAND | wxLEFT, 23);
+    m_sizer_combox->AddSpacer(FromDIP(DESIGN_LEFT_MARGIN));
 
-    auto combo_title = new wxStaticText(parent, wxID_ANY, title, wxDefaultPosition, DESIGN_TITLE_SIZE, 0);
+    auto combo_title = new wxStaticText(m_parent, wxID_ANY, title , wxDefaultPosition, DESIGN_TITLE_SIZE, wxST_NO_AUTORESIZE);
     combo_title->SetForegroundColour(DESIGN_GRAY900_COLOR);
-    combo_title->SetFont(::Label::Body_13);
+    combo_title->SetFont(::Label::Body_14);
     combo_title->SetToolTip(tooltip);
-    combo_title->Wrap(-1);
-    m_sizer_combox->Add(combo_title, 0, wxALIGN_CENTER | wxALL, 3);
+    combo_title->Wrap(DESIGN_TITLE_SIZE.x);
+    m_sizer_combox->Add(combo_title, 0, wxALIGN_CENTER);
 
-    auto                            combobox = new ::ComboBox(parent, wxID_ANY, wxEmptyString, wxDefaultPosition, DESIGN_COMBOBOX_SIZE, 0, nullptr, wxCB_READONLY);
-    combobox->SetFont(::Label::Body_13);
-    combobox->GetDropDown().SetFont(::Label::Body_13);
+    auto combobox = new ::ComboBox(m_parent, wxID_ANY, wxEmptyString, wxDefaultPosition, DESIGN_COMBOBOX_SIZE, 0, nullptr, wxCB_READONLY);
+    combobox->SetFont(::Label::Body_14);
+    combobox->GetDropDown().SetFont(::Label::Body_14);
+
+    std::vector<wxString> FlushOptions = {_L("All"), _L("Color"), _L("Filament"), _L("None")};
+    std::vector<wxString>::iterator iter;
+    for (iter = FlushOptions.begin(); iter != FlushOptions.end(); iter++) { combobox->Append(*iter); }
+
+    auto opt_color = app_config->get("auto_calculate") == "true";
+    auto opt_filam = app_config->get("auto_calculate_when_filament_change") == "true";
+    if (opt_color && opt_filam) {
+        combobox->SetValue(FlushOptions[0]);
+    }else if(opt_color){
+        combobox->SetValue(FlushOptions[1]);
+    }else if(opt_filam){
+        combobox->SetValue(FlushOptions[2]);
+    }else{
+        combobox->SetValue(FlushOptions[3]);
+    }
+
+    m_sizer_combox->Add(combobox, 0, wxALIGN_CENTER | wxLEFT, FromDIP(5));
+
+    combobox->GetDropDown().Bind(wxEVT_COMBOBOX, [this](wxCommandEvent &e) {
+        auto sel = e.GetSelection();
+        app_config->set("auto_calculate"                     ,(sel == 0 || sel == 1) ? "true" : "false");
+        app_config->set("auto_calculate_when_filament_change",(sel == 0 || sel == 2) ? "true" : "false");
+        e.Skip();
+     });
+    return m_sizer_combox;
+}
+
+wxBoxSizer *PreferencesDialog::create_item_loglevel_combobox(wxString title, wxString tooltip, std::vector<wxString> vlist)
```

Comment:
Could you also show this log level selection in dev settings too? I found it useful for debugging instead of asking peaple to modify the config file manually.

Reply:
Yep.